drivers/sensor/sensor_shell.c

```diff
@@ -299,8 +303,15 @@ static void sensor_shell_processing_callback(int result, uint8_t *buf, uint32_t
 static int cmd_get_sensor(const struct shell *sh, size_t argc, char *argv[])
 {
 	const struct device *dev;
+	int count = 0;
 	int err;
 
+	err = k_mutex_lock(cmd_get_mutex, K_NO_WAIT);
```

Comment:
There are a few return statements below that will need to unlock this mutex

Reply:
I agree, my bad. I had moved the lock to the top of the function, but originally it was placed at the very bottom. And now I have moved it back again.